bsp_diff/caas/device/intel/civ/host/vm-manager/0046-Multi-Android-VM-support.patch

```diff
@@ -0,0 +1,412 @@
++}
++
++
++function create_vms() {
++    create_vm_dirs $1 $2 $3
++    copy_files_for_vm_flashing $1 $2 $3
++}
++
++function setup_files() {
++    echo "Creating setup for VMs"
++    if [ -d $USER_HOME/.intel/.civ ]
++    then
++        SETUP_HOST_DONE=true
++    fi
++    new_dir=$CIV_DIR/$VM_NAME
++    if [ -n "$(ls -d ${new_dir}?)" ]; then
++	dir_arr=$(ls -d ${new_dir}?)
++	directories=( $dir_arr )
++	last_vm=${directories[-1]}
++	final_vm_name="${last_vm##*/}"
++	last_character=${final_vm_name: -1}
++	last_character=$((last_character+1))
++	ADB_PORT=$((5555+(last_character*2)))
++	VSOCK_ID=$((3+last_character))
++	INITIAL_VM_NUM=$last_character
++    fi
++
++    for (( i=$INITIAL_VM_NUM; i<$NUMBER_OF_VM+$INITIAL_VM_NUM; i++ ))
++    do
++	echo "Folder name: $CIV_DIR/${VM_NAME}${i}"
++
++	if [ $SETUP_HOST_DONE = false ]; then
++            ./scripts/setup_host.sh -v 3 -b $ADB_PORT -n ${VM_NAME}${i}
++	    SETUP_HOST_DONE=true
++	else
++	    ADB_PORT=$((ADB_PORT+(i*2)))
++	    create_vms $((i+3)) $ADB_PORT ${VM_NAME}${i}
++	fi
++    done
++
++    echo "Please reboot"
++}
++
++function copy_files_for_vm() {
++    echo "Copying to folder $VM_NAME$1"
++    if [ -f "$CIV_DIR/$VM_NAME$INITIAL_VM_NUM/android.qcow2" ]; then
++        cp $CIV_DIR/$VM_NAME$INITIAL_VM_NUM/android.qcow2 $CIV_DIR/$VM_NAME$1
++    else
++        echo "android.qcow2 missing. Please run setup_host.sh, start_flash_usb.sh and try again"
++        exit
++    fi
++
++    if [ -d "$CIV_DIR/$VM_NAME$INITIAL_VM_NUM/vtpm0" ]; then
++        cp -r $CIV_DIR/$VM_NAME$INITIAL_VM_NUM/vtpm0 $CIV_DIR/$VM_NAME$1
++    else
++        echo "vtpm0 missing. Please run setup_host.sh, start_flash_usb.sh and try again"
++        exit
++    fi
++}
++
++function flash_vms() {
++    echo "Flashing VMs"
++    if [ -f caas-flashfiles-*.zip ]; then
++        ./scripts/start_flash_usb.sh caas-flashfiles-*.zip --display-off -n $VM_NAME$INITIAL_VM_NUM
++    else
++        echo "flashfiles missing. Please download and unzip the package correctly."
++        exit -1
++    fi
++
++    for (( i=$INITIAL_VM_NUM+1; i<$NUMBER_OF_VM+$INITIAL_VM_NUM; i++ ))
++    do
++        copy_files_for_vm $i
++        echo "Flashed ${VM_NAME}${i}"
++    done
++    echo "Done flashing"
++}
++
++function show_help() {
++    printf "Creates specified number of Android guests under folder name Android-CIVx, where x is the guest number.\n"
++    printf "Usage: \n"
++    printf "$(basename "$0") [-c] \n"
++    printf "Options:\n"
++    printf "\t-h  show this help message\n"
++    printf "\t-c  specify number of Android guests to be newly created. Default without specifying -c, is 1.\n"
++}
++
++function parse_args() {
++    while [[ $# -gt 0 ]]; do
++        case $1 in
++	    -c)
++		NUMBER_OF_VM=$2
++		shift
++		;;
++
++	    -h|-\?|--help)
++                show_help
++                exit
++                ;;
++
++	esac
++        shift
++    done
++}
++
++parse_args "$@"
++setup_files
++flash_vms
```

Comment:
Add this same code from setup_host.sh at the end of the script file:

> echo "Done: \"$(realpath $0) $@\""

It helps to remind the user what has been executed